app/contexts/LocalAuthContext.tsx

```diff
@@ -0,0 +1,110 @@
+import { Logging } from '@api'
+import * as LocalAuthentication from 'expo-local-authentication'
+import { AuthenticationType, LocalAuthenticationOptions, SecurityLevel } from 'expo-local-authentication'
+import React, { createContext, useCallback, useContext, useEffect, useState } from 'react'
+import { PrivacyLockPersistence } from '../api/wallet/privacy_lock'
+
+export interface LocalAuthContext {
+  // user's hardware condition, external
+  fetchHardwareStatus: () => void // not likely needed, in case user change device's security setting
+  hasHardware: boolean
+  hardwareSecurityLevel: SecurityLevel
+  supportedTypes: AuthenticationType[]
+  isDeviceProtected: boolean
+
+  // API
+  isPrivacyLock: boolean
+  privacyLock: (options?: LocalAuthenticationOptions) => Promise<void>
+  enablePrivacyLock: (options?: LocalAuthenticationOptions) => Promise<void>
+  disablePrivacyLock: (options?: LocalAuthenticationOptions) => Promise<void>
+  togglePrivacyLock: (options?: LocalAuthenticationOptions) => Promise<void>
```

Comment:
I think you should reduce the interface, too much IMO.

1. support privacy lock?
2. type of supported privacy lock (choose 1)
3. isEnabled?
4. setEnabled(boolean)

Reply:
And should rename to `PrivacyLockContext`.

Reply:
Reduced API provided, but left those hardware status getter there, will be needed if UI trying to explain why privacy lock option is disabled.